### PR TITLE
Bug: Renaming tables and columns in a transaction

### PR DESCRIPTION
**Pull request by alecsammon**
When using the memory database and running migrations to rename both columns and tables this can fail depending on the order of the operations.

If the order (inside a transaction) is:
1. Rename column
2. Rename table
Then the column rename is not used in the resulting state of the database.

If the order is:
1. Rename the table
2. Rename the column
Then this is ok.

When using a true MySQL instance then both cases work correctly.

(This PR currently just contains the tests to prove the bug)

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -90,4 +90,49 @@ func TestSeverCustomListener(t *testing.T) {
...
+		require.NoError(t, err)
+
+		_, err = db.Exec("RENAME TABLE table1 TO table2")
+		require.NoError(t, err)
+
+		_, err = db.Query("SELECT id2 FROM table2")
+		require.NoError(t, err)
+	})
+
+	t.Run("should handle rename tables and columns in a transaction", func(t *testing.T) {
```
Comment (alecsammon) [inline server/server_test.go:105]: this test is currently failing. 

if inside a transaction we
1. Rename a column
2. Rename the table

Then the column rename doesn't work correctly

File: server/server_test.go [hunk 2]
```diff
@@ -90,4 +90,49 @@ func TestSeverCustomListener(t *testing.T) {
...
+		_, err = tx.Exec("ALTER TABLE table1 CHANGE `id` `id2` INT")
+		require.NoError(t, err)
+
+		_, err = tx.Exec("RENAME TABLE table1 TO table2")
+		require.NoError(t, err)
+
+		err = tx.Commit()
+		require.NoError(t, err)
+
+		_, err = db.Query("SELECT id2 FROM table2")
```
Comment (alecsammon) [inline server/server_test.go:118]: `_, err = db.Query("SELECT id FROM table2")`

This is incorrect - but does pass.

File: server/server_test.go [hunk 3]
```diff
@@ -90,4 +90,49 @@ func TestSeverCustomListener(t *testing.T) {
...
+
+	t.Run("should handle rename tables and columns in a transaction reversed", func(t *testing.T) {
+		tx, err := db.BeginTx(context.Background(), nil)
+		require.NoError(t, err)
+
+		_, err = tx.Exec("RENAME TABLE table1 TO table2")
+		require.NoError(t, err)
+
+		_, err = tx.Exec("ALTER TABLE table2 CHANGE `id` `id2` INT")
+		require.NoError(t, err)
```
Comment (alecsammon) [inline server/server_test.go:130]: In this case we rename the table first, and then alter the column - this one works as expected.

File: server/server_test.go [hunk 4]
```diff
@@ -90,4 +90,49 @@ func TestSeverCustomListener(t *testing.T) {
 	if tableName != "table1" {
 		t.Fatalf("expected to find table1, but got %s", tableName)
 	}
+
+	t.Run("should handle rename tables and columns", func(t *testing.T) {
```
Comment (alecsammon) [inline server/server_test.go:94]: When not running migrations in a transaction then there are no problems

**Comment by timsehn** [comment]
What's up here @alecsammon? Do you want us to fix the bug you identified?

**Comment by alecsammon** [comment]
Hi @timsehn  - sorry didn't link properly.
This was an example related to this issue: https://github.com/dolthub/go-mysql-server/issues/2396

This has already been fixed. I'll close this draft PR.

Thank you!

**Pull request closed without merging**